lib/taskclusterrc.js

```diff
@@ -0,0 +1,209 @@
+    source: taskclusterrc.buildConfigUrl(payload)
+  };
+};
+
+/**
+ * Uses a webhook message payload to generate a taskcluster graph config.
+ * with github specific information.
+ **/
+function buildWebHookGraphConfig(payload) {
+  let details = payload.details;
+  let githubConfig = {
+    metadata: buildWebHookConfigMetaData(payload),
+    routes: [
+      ['taskcluster-github',
+        payload.organization,
+        payload.repository,
+        details.headSha].join('.')
+    ]
+  };
+  return githubConfig;
+};
+
+/**
+ * Uses a webhook message payload to generate a taskcluster task config.
+ * with github specific information.
+ **/
+function buildWebHookTaskConfig(payload) {
+  let details = payload.details;
+  let githubConfig = {
+    metadata: buildWebHookConfigMetaData(payload),
+    extra: {
+      // The whitelist is used to determine who is allowed to trigger a job.
+      // by default it will allow public members of the base repository's
+      // organization
+      whitelist: {
+        orgs: [payload.organization],
+        users: []
+      }
+    },
+    payload: {
+      // Any environment variable with an undefined/null value will be
+      // left out of the compiled task config automatically after merging
+      env: {
+        GITHUB_EVENT: details.event,
+        GITHUB_BRANCH: details.branch,
+        GITHUB_PULL_REQUEST: details.pullNumber,
+
+        // Base details
+        GITHUB_BASE_REPO_URL: details.baseRepoUrl,
+        GITHUB_BASE_USER: details.baseUser,
+        GITHUB_BASE_SHA: details.baseSha,
+        GITHUB_BASE_BRANCH: details.branch,
+        GITHUB_BASE_REF: details.baseRef,
+
+        // Head details
+        GITHUB_HEAD_REPO_URL: details.headRepoUrl,
+        GITHUB_HEAD_USER: details.headUser,
+        GITHUB_HEAD_SHA: details.headSha,
+        GITHUB_HEAD_BRANCH: details.headBranch,
+        GITHUB_HEAD_REF: details.headRef
+      }
+    }
+  };
+  return githubConfig;
+}
+
+// Compares a list of expressions and a list of values,
+// returning true if any possible combination is a match
+function listContainsExpressions(expressions, values) {
+  for (var i in expressions ) {
+    let exp = RegExp(expressions[i], 'i');
+    // just join values so that we don't have to compare multiple times
+    if (exp.test(values.join(' '))) return true;
+  }
+  return false;
+};
+
+/**
+ * Merges an existing taskclusterrc with a pull request message's
+ * payload to generate a full task graph config. taskclusterrc
+ * is expected to be a string of yaml.
+ **/
+taskclusterrc.processConfig = function(taskclusterrc, payload, headUserInfo) {
+  return new Promise(function(accept, reject) {
+    try {
+      let taskclusterConfig = yaml.load(taskclusterrc);
+      if (!taskclusterConfig.tasks) {
+        // fall back to assuming that the user only wanted to configure a single
+        // task when no top level "tasks" key exists in the config.
+        taskclusterConfig.tasks = [taskclusterConfig];
+      }
```

Comment:
Great point; agreed. I'll use the top level version: # format you suggested to decide how the config will be parsed and add a schema for .taskclusterrc. Thanks for these comments/reviews !